contracts/Minter.sol

```diff
@@ -0,0 +1,164 @@
+    require(startBlock >= block.number, "Start block must be above current block");
+    require(newToken != address(0), 'Token cannot be address 0');
+    require(newAdmin != address(0), 'Admin cannot be address 0');
+
+    token = newToken;
+    lastMintBlock = startBlock;
+    admin = newAdmin;
+    maxPerBlock = newMaxPerBlock;
+
+    emit UpdateAdmin(address(0), newAdmin);
+  }
+
+  /**
+   * @notice Mint tokens to all pools, can be called by anyone
+   */
+  function mint() public {
+    uint256 mintDifference = block.number - lastMintBlock;
+
+    for (uint256 i = 0; i < pools.length; i++){
+      uint256 amount = pools[i].amountPerBlock * mintDifference;
+
+      if(totalMinted + amount >= cap && totalMinted != cap){
+        amount = cap - totalMinted;
+      }
+      require(totalMinted + amount <= cap, "Cap reached");
+
+      totalMinted = totalMinted + amount;
+      IMintable(token).mint(pools[i].receiver, amount);
+
+      emit Mint(pools[i].receiver, amount);
+    }
+
+    lastMintBlock = block.number;
+  }
+
+  /**
+   * @notice Add new pool, can be called by admin
+   * @param newReceiver Address of the receiver
+   * @param newAmount Amount of tokens per block
+   */
+  function addPool(address newReceiver, uint256 newAmount) external onlyAdmin {
+    require(pools.length <= poolsCap, 'Pools cap reached');
+    require(newAmount <= maxPerBlock, 'Maximum amount per block reached');
+    pools.push(Pool(newReceiver, newAmount));
+    emit PoolAdded(newReceiver, newAmount);
+  }
+
+  /**
+   * @notice Update pool, can be called by admin
+   * @param index Index in the array of the pool
+   * @param newReceiver Address of the receiver
+   * @param newAmount Amount of tokens per block
+   */
+  function updatePool(uint256 index, address newReceiver, uint256 newAmount) external onlyAdmin {
+    require(newAmount <= maxPerBlock, 'Maximum amount per block reached');
+    mint();
+    pools[index] = Pool(newReceiver, newAmount);
+    emit PoolUpdated(index, newReceiver, newAmount);
+  }
+
+  /**
+   * @notice Remove pool, can be called by admin
+   * @param index Index in the array of the pool
+   */
+  function removePool(uint256 index) external onlyAdmin {
+    mint();
+    address oldReceiver = pools[index].receiver;
+    uint256 oldAmount = pools[index].amountPerBlock;
+
+    pools[index] = pools[pools.length - 1];
+    pools.pop();
+    emit PoolRemoved(index, oldReceiver, oldAmount);
+  }
+
+  /**
+   * @notice Update admin address
+   * @param newAdmin Address of the new admin
+   */
+  function updateAdmin(address newAdmin) external onlyAdmin {
+    emit UpdateAdmin(admin, newAdmin);
+    admin = newAdmin;
+  }
+
+  /**
+   * @notice Update maximum amount that can be created per block
+   * @param newMaxPerBlock maximum number of token that is allowed to be minted per block
+   */
+  function updateMaxPerBlock(uint256 newMaxPerBlock) external onlyAdmin {
+    maxPerBlock = newMaxPerBlock;
```

Comment:
Recommend emitting an event here with old and new value

Reply:
Resolved: https://github.com/steem-monsters/sps-minter/commit/4d021c85ee8cd80ee7f45ead1121758d6a410bd0